test/unittests/execute_test.cpp

```diff
@@ -734,6 +741,14 @@ TEST(execute, memory_grow_custom_hard_limit)
         EXPECT_THAT(execute(*instance_max_limit, 0, {input}), Result(expected));
     }
 
+    {
+        bytes memory(PageSize, 0);
+        const auto instance_huge_hard_limit =
+            instantiate(*module_imported, {}, {}, {{&memory, {1, std::nullopt}}}, {}, 65536);
+        EXPECT_THAT(execute(*instance_huge_hard_limit, 0, {65536}), Result(-1));
```

Comment:
Well perhaps leave a comment then here that we should test for increasing by `65535` but don't want to spend too much time in unittests (to allocate 4gb) as spectests covers it.

Reply:
Actually I checked and there're no spec tests that allocate 4GB, only one that sets the upper limit to 4GB (but doesn't allocate anything).

Reply:
It is hard to test for it when VMs are allowed to return an error.

Reply:
Added a comment here.